src/Repositories/Behaviors/HandleFieldsGroups.php

```diff
@@ -63,12 +81,37 @@ public function getFormFieldsHandleFieldsGroups($object, $fields)
         return $fields;
     }
 
-    protected function handleFieldsGroups($fields) {
+    protected function handleFieldsGroups($fields)
+    {
         foreach ($this->fieldsGroups as $group => $groupFields) {
+            if ($this->fieldsGroupsFormFieldNameSeparator !== null) {
+                $groupFields = array_map(function ($field_name) use ($group) {
+                    return $group . $this->fieldsGroupsFormFieldNameSeparator . $field_name;
+                }, $groupFields);
+            }
+
             $fields[$group] = Arr::where(Arr::only($fields, $groupFields), function ($value, $key) {
                 return !empty($value);
             });
 
+            if ($this->fieldsGroupsFormFieldNameSeparator !== null) {
+                $fieldsGroupWithGroupSeparator = [];
+                foreach ($fields[$group] as $key => $value) {
+                    $fieldsGroupWithGroupSeparator[Str::replaceFirst($group . $this->fieldsGroupsFormFieldNameSeparator, '', $key)] = $value;
+                }
+                $fields[$group] = $fieldsGroupWithGroupSeparator;
+            }
+
+            if (in_array($group, $this->model->translatedAttributes) && is_array($fields[$group])) {
```

Comment:
```suggestion
            if (in_array($group, $this->model->translatedAttributes ?? []) && is_array($fields[$group])) {
```
I believe `?? []` or a similar check is needed for models without translations.

Reply:
> 
> 
> I believe `?? []` or a similar check is needed for models without translations.

You're right, I basically used the way test is done in the `HandleTranslations` Trait. Digging deeper, I suggest calling a function on the Model that already do the existence test instead of using directly the attribute:

```
// A17/Twill/Models/Model.php

    public function getTranslatedAttributes()
    {
        return $this->translatedAttributes ?? [];
    }
```

Do you agree with this?

Reply:
I totally agree, very clean!